### PR TITLE
Fix and enable flake8 error "imported but unused"

### PR DESCRIPTION
**Pull request by timokau**
## Description

Continuation of #104.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: csrank/__init__.py [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+from .choicefunction import __all__ as choicefunction
+from .core import __all__ as core
+from .dataset_reader import __all__ as dataset_reader
+from .discretechoice import __all__ as discretechoice
+from .objectranking import __all__ as objectranking
 from .tunable import Tunable
 from .tuning import ParameterOptimizer
+
+__all__ = ["Tunable", "ParameterOptimizer"]
+__all__.extend(choicefunction)
```
Comment (timokau) [inline csrank/__init__.py]: I'm unsure about this. We could also use `# noqa` to just skip the check here. Maybe we should reconsider whether re-exporting everything at the top level is even desirable.

**Comment by timokau** [comment]
Possibly relevant references for review:

- https://www.python.org/dev/peps/pep-0008/#public-and-internal-interfaces
- https://github.com/RaRe-Technologies/gensim/issues/1551
- https://stackoverflow.com/questions/44834/can-someone-explain-all-in-python

File: csrank/__init__.py (hunk 1 above)
Comment (timokau) [reply csrank/__init__.py]: Also looks like the current solution is not working as expected:

```
ImportError while importing test module '/home/travis/build/kiudee/cs-ranking/csrank/tests/test_choice_functions.py'.

Hint: make sure your test modules/packages have valid Python names.

Traceback:

csrank/tests/test_choice_functions.py:27: in <module>

    from csrank.tests.test_ranking import check_params_tunable

csrank/tests/test_ranking.py:8: in <module>

    from csrank import PairedCombinatorialLogit

E   ImportError: cannot import name 'PairedCombinatorialLogit' from 'csrank' (/home/travis/build/kiudee/cs-ranking/csrank/__init__.py)
```

Comment (kiudee) [reply csrank/__init__.py]: Regarding the second comment: I think in tests it is better to import using the full path anyway.

Comment (timokau) [reply csrank/__init__.py]: As discussed on slack, we might want to reduce the re-exports in the future but this is not the place to do it. So I'm going with `# noqa` comments for now.

Comment (timokau) [reply csrank/__init__.py]: Also see https://github.com/kiudee/cs-ranking/issues/105.

**Comment by timokau** [comment]
Weird, I'm pretty sure I ran the `black` check before committing.

The travis issue should be fixed now. Ready for review.

**Pull request merged**